### PR TITLE
Implement _run_and_rollback_retrying

### PR DESCRIPTION
**Pull request by msullivan**
In general we should be using retrying transactions. Our isolated
tests do retries at the outside, but non-isolated ones don't
necessarily.

Implement _run_and_rollback_retrying and use it some places that
TransactionSerializationError flakes have been seen (in #6954 and
in nightly CI).

Honestly I don't really think that these places ought to be having
TransactionSerializationErrors, but also it really is pretty required
to do retries when using SERIALIZABLE, so we should start moving
towards being more consistent about it.

### CODE REVIEW

File: edb/testbase/connection.py [hunk 1]
```diff
@@ -283,22 +283,29 @@ async def _ensure_transaction(self):
...
 
+class RawIteration(RawTransaction, Iteration):
+    async def _ensure_transaction(self):
+        self._started = True
+        await super()._ensure_transaction()
+
+
 class Retry:
-    def __init__(self, connection):
+    def __init__(self, connection, raw=False):
```
Comment (msullivan) [inline edb/testbase/connection.py:302]: Or should we make all our `Iteration`s be `RawTransaction`s?

Comment (msullivan) [reply edb/testbase/connection.py]: I ended up taking a different approach that allows Iteration to always contain nested transactions, but not to be in one

File: edb/testbase/connection.py [hunk 2]
```diff
@@ -293,7 +291,7 @@ async def _ensure_transaction(self):
 
 
 class Retry:
-    def __init__(self, connection):
+    def __init__(self, connection, raw=False):
```
Comment (fantix) [inline edb/testbase/connection.py:294]:
```suggestion
    def __init__(self, connection):
```

Unused now

**Review by fantix — APPROVED** [review]
Yeah I think this is good.

**Pull request merged**